packages/website/docs/sources.md

```diff
@@ -306,94 +465,20 @@ const source = {
 
 > `(params: { state: AutocompleteState, ...setters, event: Event, item: TItem, itemInputValue: string, itemUrl: string, source: AutocompleteSource }) => void` | defaults to `({ setIsOpen }) => setIsOpen(false)`
 
-Called when an item is selected.
+Called whenever an item is selected.
 
 ### `onActive`
 
 > `(params: { state: AutocompleteState, ...setters, event: Event, item: TItem, itemInputValue: string, itemUrl: string, source: AutocompleteSource }) => void`
 
-Called when an item is active.
+Called whenever an item is active.
 
-You can trigger different behaviors if the item is active following a mouse event or a keyboard event based on the `event` param.
+You can trigger different behaviors if the item is active depending on the triggering event using the `event` parameter.
 
 ### `templates` (specific to `@algolia/autocomplete-js`)
 
 > `SourceTemplate`
 
 The `@algolia/autocomplete-js` supports source templates.
 
-A template can either return a string, or perform DOM mutations (manipulating DOM elements with JavaScript and attaching events) without returning a string.
-
-```ts title="SourceTemplate"
-type SourceTemplate = {
-  item: Template<{
-    root: HTMLElement;
-    item: TItem;
-    state: AutocompleteState<TItem>;
-  }>;
-  header?: Template<{
-    root: HTMLElement;
-    state: AutocompleteState<TItem>;
-    source: AutocompleteSource;
-    items: TItem[];
-  }>;
-  footer?: Template<{
-    root: HTMLElement;
-    state: AutocompleteState<TItem>;
-    source: AutocompleteSource;
-    items: TItem[];
-  }>;
-};
-```
-
-```ts title="Example"
-import algoliasearch from 'algoliasearch/lite';
-import {
-  autocomplete,
-  getAlgoliaHits,
-  reverseHighlightHit,
-} from '@algolia/autocomplete-js';
-
-const searchClient = algoliasearch(
-  'latency',
-  '6be0576ff61c053d5f9a3225e2a90f76'
-);
-
-const autocompleteSearch = autocomplete({
-  container: '#autocomplete',
-  getSources() {
-    return [
-      {
-        getItemInputValue({ item }) {
-          return item.query;
-        },
-        getItems({ query }) {
-          return getAlgoliaHits({
-            searchClient,
-            queries: [
-              {
-                indexName: 'instant_search_demo_query_suggestions',
-                query,
-                params: {
-                  hitsPerPage: 4,
-                },
-              },
-            ],
-          });
-        },
-        templates: {
-          header() {
-            return 'Suggestions';
-          },
-          item({ item }) {
-            return reverseHighlightHit({ hit: item, attribute: 'query' });
-          },
-          footer() {
-            return 'Footer';
-          },
-        },
-      },
-    ];
-  },
-});
-```
+A template can either return a string or perform DOM mutations without returning a string.
```

Comment:
This is the sentence from the [source code](https://github.com/algolia/autocomplete.js/blob/next/packages/autocomplete-js/src/types/AutocompleteSource.ts#L14) but I find it confusing.

What do you mean by "perform DOM mutations without returning a string"? If you don't return anything, the template will be empty, won't it?

Reply:
This is not true anymore from #381 (which will get merged soon once coordinated with other open PRs).

As documented above, we now support strings, components, etc. so you can ignore this outdated formulation.

Reply:
Handled here 2ce1218.